src/util/pointabilities.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+Minetest
+Copyright (C) 2023 cx384
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "pointabilities.h"
+
+#include "serialize.h"
+#include "exceptions.h"
+#include <sstream>
+
+PointabilityType Pointabilities::deSerializePointabilityType(std::istream &is)
+{
+	PointabilityType pointable_type = static_cast<PointabilityType>(readU8(is));
+	switch(pointable_type) {
+		case PointabilityType::POINTABLE:
+		case PointabilityType::POINTABLE_NOT:
+		case PointabilityType::POINTABLE_BLOCKING:
+			break;
+		default:
+			pointable_type = PointabilityType::POINTABLE;
+			break;
+    }
```

Comment:
Looks like some spaces crept in here, should be tabs.

Reply:
fixed